ext/json/json_parser.tab.c

```diff
@@ -1882,12 +1997,17 @@ int php_json_parser_object_update(php_json_parser *parser, zval *object, zend_st
 		ZVAL_NEW_STR(&zkey, key);
 		zend_std_write_property(object, &zkey, zvalue, NULL); 
 
-		if (Z_REFCOUNTED_P(zvalue)) {
-			Z_DELREF_P(zvalue);
-		}
+		zval_dtor(zvalue);
 	}
 	zend_string_release(key);
 
+	if (EG(exception)) {
+		parser->scanner.errcode = PHP_JSON_ERROR_DEPTH;
```

Comment:
Thinking about it, I would actually go for a new json error as the depth is for something else so user knows that it failed because of the attack.